src/Core/src/Dispatching/Dispatcher.Windows.cs

```diff
@@ -69,6 +72,20 @@ public void Start()
 			if (IsRunning)
 				return;
 
+			if (_started)
+			{
+				// if we Start a timer that has already finished, it will immediately fire.
+				// to ensure that Start restarts the timer, we need to replace it.
+				var newTimer = _dispatcherQueue.CreateTimer();
+				newTimer.Interval = Interval;
+				newTimer.IsRepeating = IsRepeating;
+				_timer = newTimer;
+			}
+			else
+			{
+				_started = true;
+			}
+
 			_timer.Tick += OnTimerTick;
```

Comment:
I think this is because we keep attaching the event handler every time we start. Maybe we can attach this event handler in the ctor? And then removed the detach from the Stop?

Or we can always create the timer in the Start and then clean up in Stop, but that might be more overhead. Not sure how heavy the timer is.